### ISSUE TITLE
factor ComboBoxListBox out of ComboBox

### ISSUE THREAD

**Issue by pixelzoom** [phetsims/sun#445]
In https://github.com/phetsims/sun/issues/430, I was unable to factor "list box" out of ComboBox.  After hours of working on this, I bailed. According to https://github.com/phetsims/sun/issues/314#issuecomment-451568406, the code is "not in an acceptable state". The a11y code is too intertwined with the implementation, not adequately documented, and I had no idea what I might be breaking.  

Highly recommended to complete this refactor before further a11y work in #314.

FYI, the name `ComboBoxListBox` was chosen by consensus in https://github.com/phetsims/sun/issues/440.

Assigning to @ariel-phet to prioritize and assign.

**Comment by ariel-phet** [comment]
@zepumph can you give me some insight as to an estimate of how much time it will take to get this code into an acceptable state?

In discussing with @pixelzoom one option may well be to remove the current a11y implementation and start from scratch. He was not necessarily advocating for that approach, but said that it might be easier depending on where you feel the a11y design started and where your understanding of it is now. My understanding is the design is solid, but the implementation currently leaves a bit to be desired. 

If this is a more appropriate question for @jessegreenberg please assign to him.

Either way, once you have an estimate, also let me know if there is time to work on this in the near future, and then assign back to me. Considering this UI element is fairly ubiquitous to sims, it seems fairly high priority to have it in good shape.

**Comment by zepumph** [comment]
I would say my estimate would be 2-3 days of work. Much of this would involve reworking the a11y implementation, and to me that feels fine. Again what is currently in place was a prototype, and should go through some refining no matter what.

There are 12 TODOs relating to the a11y code that will be worked on as part of this, and likely others too.

Finally it would be good to note that making ComboBox accessible and ready for production is in line with a11y goals too, because Molarity is one of the next sims to have description work done with it. I hope to work on ComboBox heavily in the next 2 weeks or so.

**Comment by pixelzoom** [comment]
I'd be happy to help with this.  Since "what is currently in place was a prototype", my recommendation would be to first create a solid implementation, then add fresh a11y instrumentation to it.

**Comment by ariel-phet** [comment]
@zepumph sounds good....and glad to know this is all in line with current a11y goals.

Please schedule time to work with @pixelzoom when this work rises to the top of your work pile. If you are open to it, I would go with his suggestion of adding fresh a11y instrumentation.

**Comment by pixelzoom** [comment]
@zepumph I'm going to take a quick first pass at `ComboBoxListBox`.

**Comment by pixelzoom** [comment]
@zepumph After you've had a look at `ComboBoxListBox`, I think the next step would be to evaluate whether there's any code in `ComboBox` that belongs in either `ComboBoxListBox` or `ComboBoxButton`.

**Comment by pixelzoom** [comment]
I noticed that KEY_ENTER selected items in the listbox, but does not pop down the listbox. In the above commit, I added a TODO to investigate.

**Comment by pixelzoom** [comment]
I finally understand that setting focus should only be done in a11y specific listeners that handle keystrokes.   Fixed applied in https://github.com/phetsims/sun/commit/a484df889bd92bab290b54bf478ba5fcf2f1a1ae so that focus highlights don't appear when using the mouse.

**Comment by pixelzoom** [comment]
@zepumph and I spent several hours cleaning up TODO items for this and other ComboBox issues. All TODO items related to this issue have been addressed or moved to other issues.  So I'm comfortable with closing this issue.  @zepumph if you agree, please close.
